vllm/lora/layers.py

```diff
@@ -443,7 +449,8 @@ def can_replace_layer(
         packed_modules_list: List,
         model_config: Optional[PretrainedConfig],
     ) -> bool:
-        return type(source_layer) is ReplicatedLinear
+        return type(source_layer) is ReplicatedLinear or _is_hf_linear(
+            source_layer, ReplicatedLinear)
```

Comment:
```suggestion
        # Check parent_cls in case source_layer is a HFCompatibleLinear
        layer = getattr(source_layer, "parent_cls", source_layer)
        return type(layer) is ReplicatedLinear
```

Reply:
This part could be done using `super` to get `layer` to avoid duplicating the `getattr` for all classes